### ISSUE TITLE
National Geospatial Data Asset (NGDA) Metadata Guidelines 

### ISSUE THREAD

**Issue by mwengren** [OSGeo/Cat-Interop#27]
The PDF published on GeoPlatform.gov has recommendations for web service metadata for inclusion in their own catalog (see PDF here: https://cms.geoplatform.gov/sites/default/files/document_library/NGDA_Metadata_Guidelines.pdf).

This recommends including a string in <gmd:applicationProfile> with a URI for a service type.  Does this conflict with the recommendation here?  Below is an example CI_OnlineResource from the NGDA PDF: 

``
<gmd:CI_OnlineResource>
<gmd:linkage>
<gmd:URL>http://maps.ngdc.noaa.gov/arcgis/services/web_mercator/trackline_combined_dynamic/MapServer/WMSServer?request=GetCapabilities&amp;service=WMS</gmd:URL>
</gmd:linkage>
<gmd:protocol>
<gco:CharacterString>http</gco:CharacterString> (optional)
</gmd:protocol>
<gmd:applicationProfile>
<gco:CharacterString>http://opengis.net/spec/wms</gco:CharacterString>
</gmd:applicationProfile>
<gmd:name>
<gco:CharacterString>Geophysical Trackline Data Web Map Service (WMS)</gco:CharacterString>
</gmd:name>
<gmd:description>
<gco:CharacterString>Capabilities document for Open Geospatial Consortium Web Map Service for Geophysical Trackline Data</gco:CharacterString>
</gmd:description>
<gmd:function>
<gmd:CI_OnLineFunctionCode codeList="http://www.ngdc.noaa.gov/metadata/published/xsd/schema/resources/Codelist/gmxCodelists.xml#CI_OnLineFunctionCode" codeListValue="search">search</gmd:CI_OnLineFunctionCode>
</gmd:function>
</gmd:CI_OnlineResource>
``

Some of the URIs in the NGDA PDF seem to differ with these: https://github.com/OSGeo/Cat-Interop/blob/master/LinkPropertyLookupTable.csv, but is the overall recommendation here of what field value from the lookup table should go in what element in an ISO 19115 metadata record compatible with the NGDA recommendation?  In other words, what is the recommendation for where 'Identifier' field value should go in ISO 19115.  NGDA seems to recommend strict use of <gmd:protocol> (eg. 'http', 'ftp'). 

Seems like it would be good to have some agreement between both recommendations if it doesn't already exist.

**Comment by tomkralidis** [comment]
@mwengren the CSV lookup's `identifier` field is what is intended to populate (for ISO) `gmd:protocol`.  Having said this it looks like this has been partly addressed in #28?

**Comment by mwengren** [comment]
@tomkralidis @jjediny I think the NGDA recommendation is conflicting with the intention here then.  

From what I can tell from [this](https://cms.geoplatform.gov/sites/default/files/document_library/NGDA_Metadata_Guidelines.pdf) doc, it seems to suggest an alternate way to achieve the same goal of characterizing service types in ISO (and CSDGM but I'm more concerned about the ISO case).  

Using the LinkPropertyLookupTable.csv, following what you said you could include something like the following for WMS?  I realize you're not recommending using applicationProfile necessarily, but including it in addition to protocol goes part way towards matching the NGDA guidelines, but not completely (see below).

```
<gmd:protocol>
<gco:CharacterString>OGC:WMS</gco:CharacterString> (optional)
</gmd:protocol>
<gmd:applicationProfile>
<gco:CharacterString>http://www.opengis.net/def/serviceType/ogc/wms</gco:CharacterString>
</gmd:applicationProfile>
```

The NGDA guidelines are suggesting a stricter use of the protocol field for transfer protocol (http|https|ftp), and using applicationProfile to indicate service type/service type compliance.  Here's an example from the doc:
```
<gmd:protocol>
<gco:CharacterString>http</gco:CharacterString> (optional)
</gmd:protocol>
<gmd:applicationProfile>
<gco:CharacterString>http://opengis.net/spec/wms</gco:CharacterString>
</gmd:applicationProfile>
```

I like the broader use of protocol you're going for, but it seems that there's some conflicting approaches here that will cause implementation differences - at least for those to whom NGDA guidelines apply.  

The NGDA guidelines will no doubt dictate to some degree how things are done here in the US if followed, I was just hoping to raise the issue in case these could be adapted to be more compatible with each other to prevent that from happening...

**Comment by smrgeoinfo** [comment]
The NGDA guidelines are very disappointing. Any the ISO link value is required to be a URI and as we know, links (URLs) start with a prefix that specifies the protocol. Including this again in a separate field is redundant.  Its more useful to specify the protocol that is tunneling on http in gmd:protocol, e.g. ogc:wms, ogc:wfs, opendap, odata, etc.  For actual interoperability, the user needs to know some additional information in many cases, e.g. with a WFS, what is the application schema for the features offered?  This is the kind of information that is most useful to put in application profile.  Guess the NGDA didn't look the cat-interop wiki... see https://github.com/OSGeo/Cat-Interop/wiki/Link-Properties-Proposal for more information.

**Comment by rsignell-usgs** [comment]
@mwengren , are you part of the NGDA team?  
 If so, perhaps you can communicate these issues back to them.    It sure would be nice to be on the same page!

**Comment by mwengren** [comment]
@rsignell-usgs No, I am not, but I think @jjediny may be.... or maybe he knows the best person to forward this to.  If not, I can probably raise it through some NOAA channels to hopefully get up to that level.

**Comment by mwengren** [comment]
@rsignell-usgs No, I am not, but I think @jjediny may be.... or maybe he knows the best person to forward this to.  If not, I can probably raise it through some NOAA channels to hopefully get up to that level.

**Comment by smrgeoinfo** [comment]
I'd be glad to send a comment on the NGDA recs, I'm just not clear on who it would go to

**Comment by pvgenuchten** [comment]
Here in the Netherlands we have a requirement to fill the gmd:protocol with a value from a predefined codelist in the national metadata profile. However we have not been able to extend that requirement into european inspire regulations, at that level discussion is still continuing...
http://www.geonovum.nl/sites/default/files/Nederlands%20metadata%20profiel%20op%20ISO%2019115%20voor%20geografie%20-%20v1.3.1%20def.pdf page 42 (in dutch)
